src/script/api/script_industry.hpp

```diff
@@ -246,7 +246,7 @@ class ScriptIndustry : public ScriptObject {
 	 * @param industry_id The index of the industry.
 	 * @param cargo_type The cargo to query, or CT_INVALID to query latest of all accepted cargoes.
 	 * @pre IsValidIndustry(industry_id).
-	 * @pre IsValidCargo(cargo_type) || cargo_type == CT_INVALID.
+	 * @pre IsValidCargo(cargo_type) || !IsValidCargoID(cargo_type).
```

Comment:
This looks odd. Requiring a cargo to be either valid or invalid... Maybe the old comment was better, to specify that if it was invalid, it was a known value instead of...what else would it be? Undefined?

Reply:
Yeah, it's weird and makes no mention of a 3rd state that is neither valid or invalid... Dunno!

Reply:
Look at line 247. CT_INVALID is a special case in the API. So replacing this with `!IsValidCargoID(cargo_type)` is not the same, as it also allows other garbage that isn't `CT_INVALID`. After all, the script can pass any (integer) garbage as parameter.

Reply:
`CT_INVALID` works well for a special case, and changing stuff like that in the API would need compatibility layers everywhere.

Reply:
I've reverted this documentation change. The behaviour was never changed.